pkg/kubelet/podcertificate/podcertificatemanager.go

```diff
@@ -476,6 +498,24 @@ func (m *IssuingManager) handleProjection(ctx context.Context, key projectionKey
 
 		klog.V(4).InfoS("Time to refresh", "key", key)
 
+		// The current time is more than 10 minutes past the most recently issued certificate's `beginRefreshAt` timestamp but the state has not been labeled with overdue for refresh.
+		if m.clock.Now().After(state.beginRefreshAt.Add(refreshOverdueDuration)) && !state.overdueForRefresh {
```

Comment:
Doesn't `credStateFresh` imply `state.overdueForRefresh` would change? Do we need that bool in the struct? Same for `state.expired`